### ISSUE TITLE
Pin doesn't work when scrambled (Pixel 3 XL)

### ISSUE THREAD

**Issue by TempUser13** [GrapheneOS/os-issue-tracker#660]
I have a Pixel 3 XL on the latest build at writing (RQ3A.210905.001.2021091407) and if I enable the Scramble Pin option, I am unable to unlock my phone through pin (fingerprint still works). If I deselect the scramble pin option, I can unlock my phone again.

**Comment by thestinger** [comment]
Not able to replicate this.

**Comment by thestinger** [comment]
You're going to need to provide a clear explanation of what doesn't work. Does the scrambled PIN menu show up and are you able to enter the PIN and attempt unlock? What exactly doesn't work? Are you correctly entering the PIN? What error message do you get?

**Comment by TempUser13** [comment]
Sorry for not being clear in the beginning. If I have scrambled pin enabled, I turn my screen on, swipe up to bring the enter pin screen and when I type my pin in, it will say "wrong pin". If I enter the exact same pin with scramble off, its accepted.

**Comment by thestinger** [comment]
Is the PIN keypad being visibly scrambled? It's unclear why this wouldn't be working for just you while it works fine for everyone else. It doesn't really make sense.

**Comment by Zanthed** [comment]
Can't reproduce this either.
Pixel 5, registered a fingerprint to unlock and enabled PIN scrambling. Both work just fine. Still works even without registering a fingerprint.

**Comment by thestinger** [comment]
You'll need to provide working instructions for reproducing the problem. It's currently a support issue because it could be caused by a change that you've made with a device manager, etc. Not able to provide that kind of support here.

**Comment by etuttle** [comment]
I'm keenly interested in this issue as well.  I'm the guy who changed his pin and enabled scrambling (note: not sure what order I did those two things).  Then I rebooted to find I could not unlock.  Then I used the adb command to disable pin scrambling, rebooted again, then what I remembered to be my new pin did not work.  Since then all I've done is collect as much as I can from adb logs, and shut the phone down.  I'm not going to give up easy because, let's just say I have valuable data on the device.  So it's worth taking my time to make the best possible guesses.  Given I've already used up 16, I believe.  I'm about to see what I can learn from running an emulator.
